### DISCUSSION TITLE
Zoom out / Zoom in for Mobile players

### DISCUSSION THREAD

**Discussion by Z1kou** [sunflower-land/sunflower-land#1208] [Ideas]
*We at Pixelized Crib are exploring new features to add to the game, and today I am gonna open several discussion's for every single one of them so sit down get a drink and buckle up because i need your spicy opinion's everyone !**

 # ISSUE

As a mobile user, finding the goblin always give me a headache. Over time I know where the goblin will usually be located, but it still takes time to search through the map. It's quite troublesome if I want to find the goblin quickly when I am busy on other stuff (eg. on the streets)

# SOLUTION

Adding a Zoom out / in Feature for mobile players the most ideal / common one would be by the pinching in / out of fingers indicating zooming out / in that is both intuative and mobile player friendly, potentially adding also an arrow pointing to the goblins as this only helps legitimate players and a huge  quality of life change for everyone that struggles with finding the goblin while ensuring that it is a functioning botting system !

What are your thoughts guys?

**Comment by Aminbtdds** [comment]
Great idea, specially on low res. Screens it is quite hard to find it👌

**Comment by JustJayHSP** [comment]
Good. If we cant have a zooming feature through moving two fingers closer or farther.

We can have a button instead of [ - ] and [ + ] to help with the zoom in and out feature. Makes it more fixed if the coding for finger zooms wont work.

**Comment by Droidze** [comment]
It would be great if the zoom in/out feature add to the game, most of the time it's really painful to find the goblin on a small screen.

**Comment by sabryder** [comment]
It would be great if the zoom in/out feature add to the game.
Great Idea😍
i hate that goblin btw😂😂💔

**Comment by ShiningMinions** [comment]
I have a few suggestions here. But first, instead of discussing pinch-zoom feature, I want us to take a step back and hyper-focus on the **problem statement**:

### PROBLEM STATEMENT:
**"On mobile, finding the goblin always give me a headache. It's very hard to find."**

Can pinch-zoom solve this problem? Definitely. However, I really think we should explore the problem a bit more, because there might be an even better solution that we're not seeing.

### Challenge Question:

Before we even think about pinch-zoom, perhaps we should be talking about the whole "goblin stealing shovel" mechanic of the game. Because if we remove/alter this mechanic altogether, then there's no reason for pinch-zoom solution in the first place.

So here's my question... Is the goblin stealing your shovel even a good mechanic to begin with? Does the benefit of stopping botters outweigh the inconvenience to the player? I cannot see the back-end logic of the game, but I would argue that this mechanic is adding very little value for stopping bots, and is adding much inconvenience for the player.

### Changing the Goblin Stealing Mechanic:

What if instead of adding pinch-zoom feature, we **change the mechanic of the goblin stealing shovel**? How about instead of searching the map for the goblin, we do something else? My suggestion: implement a slider puzzle captcha modal, similar to Binance. Perhaps the Modal UI of this could be a "farmer", and you need to slide him over to the "goblin who has your shovel". No searching the map required - everything is contained within the modal, just like reward seeds modal. Therefore, there is no longer a problem for mobile players:

![binance-captcha](https://user-images.githubusercontent.com/103600068/178508154-dd380458-e163-4964-b5f5-3776ca0e0633.png)

### Alternative Solutions:

If we cannot change the goblin-stealing mechanic, then perhaps one of these alternative solutions would help mobile players:

1. Have a UI arrow, which points to the direction of the goblin who stole your shovel.

**Reply by Danverse98** [reply]
I think the main point here is that mobile players should have the same efficiency of gameplay as the computer ones, it's really simple imo. Adding a captcha to almost every choke point that a bot may experience just takes the ordinary player out of the immersion the game offers because now he has a duty to prove he's not a bot many times per day, with a goblin stealing shovel there's atleast the immersion in the game(not the best, but annoying the most for mobile players, pc players can do it in 5 seconds) while offering an anti-bot feature at the same time. Why shouldn't we enable a zoom in/out feature for mobile when PC already have it?

**Reply by hujikol** [reply]
Quick concept on the UI arrow that I've been proposed a long time ago

<image src="https://user-images.githubusercontent.com/12151051/180106176-7b681eba-84db-4517-9ace-cc9449ea6033.jpg" width="540">
<image src="https://user-images.githubusercontent.com/12151051/180106182-e2737a75-eea0-404a-b77e-07ff55c04f3c.jpg" width="540">
<image src="https://user-images.githubusercontent.com/12151051/180106186-60f96689-46e1-43d3-99a4-2ae44296f3a3.jpg" width="540">
<image src="https://user-images.githubusercontent.com/12151051/180106188-5363b7b2-b46f-44de-b7f4-d54d5a2c3bed.jpg" width="540">

The arrow needs to be redesigned to be more in contrast to the map.

**Reply by Polysemouse** [reply]
We need to determine the effectiveness of the current shovel stealing feature to see if it's worth switching over to another anti-bot mechanism.  Adding arrows might allow bots to find the shovel goblin easier, so that's something we have to consider when we add an arrow pointing to the goblin.

I've heard that there's a way for bots to bypass a captcha, not sure if I'm right or not
